README.md

```diff
@@ -1,11 +1,86 @@
 # README
 
-Add your handler readme here. Remember to include the following:
+## What is this?
+`sceptre-sam-handler` is a TemplateHandler for Sceptre (versions 2.7 and up) that lets you use an
+AWS SAM template (and its associated project) as a stack's template.
 
-- Tell people how to install it (e.g. pip install ...).
-- Be clear about the purpose of the template handler, its capabilities and limitations.
-- Tell people how to use it.
-- Give examples of the template handler in use.
+This template handler will run `sam build` and then `sam package` from the indicated SAM Template's
+directory in order to generate a CloudFormation-ready template.
 
-If you have any questions or encounter an issue
-[please open an issue](https://github.com/Sceptre/project/issues/new)
+## How to install sceptre-sam-handler
+
+Simply `pip install scepre-sam-handler`!
+
+## How to use sceptre-sam-handler
+
+The template "type" for this handler is `sam`.
+
+This handler takes several arguments, two of which are required.
+
+### Arguments:
+* `path` (string, required): The path **from the current working directory** (NOT the
+* project path) to the SAM Template.
+* `artifact_bucket_name` (string, required): The bucket name where artifacts should be uploaded to
+on S3 during the packaging process. If your project has a `template_bucket_name`, you can set this
+to `{{ template_bucket_name }}`.
+* `artifact_prefix` (string, optional): The prefix to apply to artifacts uploaded to S3. This can be
+the project's `{{ template_key_prefix }}`.
+* `build_args` (dict, optional): Additional key/value pairs to supply to `sam build`. For
+flag-type arguments that have no value, set the value to "True".
+* `package_args` (dict, optional): Additional key/value pairs to apply to `sam package`. The
+same is true here as for `build_args` for flag-type arguments.
+
+### Default behavior
+SAM commands are invoked using the system shell in a subprocess, with stdout redirected to stderr.
+Artifacts will be uploaded using the `artifact_bucket_name` and `artifact_prefix` arguments, the
+`project_code`, and the Sceptre stack name.
+
+For example, given an `artifact_bucket_name` of "bucket", `artifact_prefix` of "prefix", a
+`project_code` of "project" and a stack config located at "config/indigo/sam-application.yaml", SAM
+artifacts will be uploaded to:
+
+`s3://bucket/prefix/project/indigo/sam-application/sam_artifacts/`
+
+By default, these will be the sam commands that are run _from the template's directory_:
+```shell
+sam build --cached --template-file [path as absolute path]
+sam package \
+  --s3-bucket [artifact_bucket_name argument] \
+  --region [the stack region] \
+  --s3-prefix [the prefix described above] \
+  --template-file [path as absolute path]
```

Comment:
I think it would be beneficial to also deploy the transformed (output) template as well.  although i think that can be done as an improvement.

Reply:
Sceptre will do that for you if you've used a `template_bucket_name`. No need to bother doing that in the handler.